Library/Homebrew/extend/os/linux/requirements/java_requirement.rb

```diff
@@ -0,0 +1,19 @@
+require "language/java"
+
+class JavaRequirement < Requirement
+  default_formula "jdk"
+
+  env do
+    next unless @java_home
+    env_java_common
+    if (Pathname.new(@java_home)/"include").exist? # Oracle JVM
```

Comment:
Suggestion: just make this `env_oracle_jdk` and skip `next unless @java_home` in favour of changes below.